### ISSUE TITLE
Sometimes estimation labels don't work

### ISSUE THREAD

**Issue by fedorpashin** [self-xdsd/self-core#1227]
I've just created an [issue](https://github.com/fedorpashin/physics/issues/21) with the label `30 min`, but the PM estimated it by default time. In other [issue](https://github.com/fedorpashin/physics/issues/11) with custom estimation everything worked well.

**Comment by zoeself** [comment]
@fedorpashin thank you for reporting this. I'll assign someone to take care of it soon.

**Comment by zoeself** [comment]
@amihaiemil I couldn't find any assignee for this task. This is either because there are no contributors with role ``DEV`` available or because the project does not have enough funds.

Please, make sure there is at least one available contributor with the required role and the project can afford to pay them.

**Comment by amihaiemil** [comment]
@fedorpashin when a new Issue is opened, Github will send us a JSON representing the Issue, also containing the Labels.

Now, I noticed Github uses event-sourcing: sometimes certain events like labeling, assigning etc are happening in different orders. Quite rarely, but I've noticed this happens.

In this particular case, I see nothing in the logs, so I think Github sent us the New Issue event before actually appending the Label event.

Also, I have never faced this problem. Did it happen more times to you or this is the first time?

**Comment by zoeself** [comment]
@amihaiemil this is your task now, please go ahead. Deadline (when this ticket should be closed) is ``2021-10-06T14:55:08.156451``.

Estimation here is ``60 minutes``, that's how much you will be paid.

Remember, you don't have to solve everything in this ticket - you can solve it [partially](https://docs.self-xdsd.com/microtasking.html) and leave ``todo`` markers in the code, which will become future tasks.

If you have any questions don't ask me, I'm not a technical person. Open new tickets instead.

If you don't have time or simply don't want to work on this, you can always [resign](https://docs.self-xdsd.com/projectmanager.html#commands).

**Comment by fedorpashin** [comment]
@amihaiemil yes, it happened one more time (the [issue](https://github.com/fedorpashin/physics/issues/6) is deleted now). Maybe I click "Submit new issue" too quickly in these cases and that's why the order is broken. I think it would be great if the PM checks labels at the moment of assigning, because sometimes I reconsider something immediately after issue creating and want to change the estimation. It's intuitive to have this opportunity in the time interval between creating and assigning)

**Comment by amihaiemil** [comment]
@fedorpashin yep, the event disorder also happens to me sometimes when I click submit too quickly. But still, rarely.

I'll consider your advice and try to implement it. I think most likely the PM will listen to issue labeling (so as you said, you will be able to change estimation after issue is refistered).

In the meantime let's leave this issue opened for future reference.

**Comment by zoeself** [comment]
@amihaiemil Don't forget to close this ticket before the deadline (2021-10-06T14:55:08). You are past the first half of the allowed period.

**Comment by zoeself** [comment]
@amihaiemil Looks like you've missed the task deadline (2021-10-06T14:55:08). You are now resigned from this task.

Please stop working on it, you will not be paid. I will assign it to someone else soon.

**Comment by zoeself** [comment]
@amihaiemil I couldn't find any assignee for this task. This is either because there are no contributors with role ``DEV`` available or because the project does not have enough funds.

Please, make sure there is at least one available contributor with the required role and the project can afford to pay them.
